app/controllers/apps_controller.rb

```diff
@@ -40,6 +38,13 @@ def destroy
   end
   private
 
+  def app_owner?
+    if !App.find(params[:id]).owners.include?(current_user)
```

Comment:
@case-eee Hmm. This is an authorization control before_action method to keep unauthorized users away from show, update, delete actions for apps they don't own. Is this something that can be controlled farther down the stack? I can push it to ApplicationController if need be, but not sure how to get it out of a controller altogether/